src/opm/io/eclipse/ESmry_write_rsm.cpp

```diff
@@ -0,0 +1,198 @@
+#include <ostream>
+#include <regex>
+#include <string>
+
+#include <opm/common/ErrorMacros.hpp>
+
+namespace {
+
+    constexpr std::size_t column_width { 8 } ;
+    constexpr std::size_t column_space { 5 } ;
+
+    constexpr std::size_t column_count { 10 } ;
+
+    constexpr std::size_t total_column { column_width + column_space } ;
+    constexpr std::size_t total_width  { total_column * column_count } ;
+
+    const std::string version_line { } ;
+    // the fact that the dashed header line has 127 rather than 130 dashes has no provenance
+    const std::string divider_line { std::string(total_width - 3, '-') } ;
+
+    const std::string block_header_line(const std::string& run_name, const std::string& comment) {
+        return "SUMMARY OF RUN " + run_name + " OPM FLOW VERSION 1910 " + comment;
+    }
+
+    void write_line(std::ostream& os, const std::string& line, char prefix = ' ') {
+        os << prefix << std::setw(total_width) << std::left << line << '\n';
+    }
+
+    void print_text_element(std::ostream& os, const std::string& element) {
+        os << std::setw(8) << std::left << element << std::setw(5) << "";
+    }
+
+    void print_float_element(std::ostream& os, float element) {
+        static const std::regex integer_regex { "\\.0*$" };
+
+        auto element_string = std::to_string(element);
+        element_string = std::regex_replace(element_string, integer_regex, "");
+
+        if (element_string.size() > 8) {
+            element_string = element_string.substr(0, 8);
+        }
+
+        os << std::setw(8) << std::right << element_string << std::setw(5) << "";
+    }
+
+    void write_header_columns(std::ostream& os, const std::vector<Opm::EclIO::SummaryNode>& vectors, std::function<void(std::ostream&, const Opm::EclIO::SummaryNode&)> print_element, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : vectors) {
+            print_element(os, vector);
+        }
+
+        os << '\n';
+    }
+
+    void write_data_row(std::ostream& os, const std::vector<std::pair<std::vector<float>, int>>& data, std::size_t index, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : data) {
+            print_float_element(os, vector.first[index] * std::pow(10.0, -vector.second));
+        }
+
+        os << '\n';
+    }
+
+    void write_scale_columns(std::ostream& os, const std::vector<std::pair<std::vector<float>, int>> data, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : data) {
+            const auto scale_factor { vector.second } ;
+            if (scale_factor) {
+                print_text_element(os, "*10**" + std::to_string(scale_factor));
+            } else {
+                print_text_element(os, "");
+            }
+        }
+
+        os << '\n';
+    }
+
+}
+
+namespace Opm::EclIO {
+
+void ESmry::write_block(std::ostream& os, const std::vector<SummaryNode>& vectors) const {
+    write_line(os, version_line, '1');
```

Comment:
We don't know - but personally I do not think this '1' is a version indicator - but rather just a block separator. I don't think the variable should be called something with *version* - I think it is misleading.